pkg/tuf/client.go

```diff
@@ -105,9 +105,14 @@ func (c *Client) loadMetadata() error {
 	} else if c.opts.CacheValidity > 0 {
 		cfg, err := LoadConfig(c.configPath())
 		if err != nil {
-			// Config may not exist, don't error
-			// create a new empty config
-			cfg = &Config{}
+			// Config will not exist during the first initialization
+			// of unexpired metadata
+			cfg = &Config{
+				LastTimestamp: time.Now(),
```

Comment:
I haven't tried yet, but does this mean that the client will assume that my local cache is already fresh, without first checking `~/.sigstore/root/tuf-repo-cdn.sigstore.dev.json`?

Reply:
If the cache is already fresh, meaning `c.up.Refresh()` succeeded, and no configuration file exists, then no additional refresh will occur. This addition persists the current time so that on the next metadata load, the config will be checked.

Reply:
By fresh, I meant that my local copy up-to-date **before** creating the client.

Reply:
`c.up.Refresh` when `UnsafeLocalRefresh` is set does not make any network calls, it simply loads the metadata from disk. It will err out if the timestamp is out of date, that we cannot and should not bypass.

Does that answer your question?

Reply:
Those options seem like the right thing I could use. Would you make those publicly accessible?

Reply:
I don't think we should expose `UnsafeLocalRefresh`, but instead make sure we have caching options that match expected use cases. Which I think we do have currently, besides what's in this PR.

The more I look at this, I think that `ForceCache` is what most users should be using, and that `CacheValidity` doesn't have a clear use case.